src/pykeen/nn/init.py

```diff
@@ -223,6 +236,13 @@ def __call__(self, x: torch.Tensor) -> torch.Tensor:
             raise ValueError(f"shape does not match: expected {self.tensor.shape} but got {x.shape}")
         return self.tensor.to(device=x.device, dtype=x.dtype)
 
+    def as_embedding(self, **kwargs: Any) -> "Embedding":
+        """Get a static embedding from this pre-trained initializer."""
+        from .representation import Embedding
+
+        max_id, *shape = self.tensor.shape
+        return Embedding(max_id=max_id, shape=shape, initializer=self, trainable=False, **kwargs)
+
```

Comment:
what about moving that to another base-class / mixin? then we can keep WL as it was before

Reply:
I had thought about doing another base class too! Maybe we can figure out https://github.com/pykeen/pykeen/pull/979#discussion_r901159700 first then this might become more obvious